src/segmantic/seg/dataset.py

```diff
@@ -75,27 +68,39 @@ def validation_files(self) -> Sequence[Dict[str, Path]]:
     def _create_split(
         self,
         data_dicts: List[Dict[str, Path]],
-        valid_split: float,
-        shuffle: bool,
-        random_seed: int = None,
-        max_files: int = 0,
+        valid_split: Optional[float] = None,
```

Comment:
I am not sure why you added `Optional` to all these arguments. `Optional` does not mean *this option is only optional*. `Optional[X]` is equivalent to `X | None` (or `Union[X, None]`).

But when option implicitly/explicitly has a default, it is more explicit to just write `shuffle: bool = True` or `valid_split: float = 0.2`, etc.

Reply:
Yep, I misunderstood the function of this. Removed it again. Thank you